packages/plans-grid-next/src/hooks/data-store/use-grid-plans-for-comparison-grid.ts

```diff
@@ -2,39 +2,89 @@ import {
 	FeatureList,
 	PLAN_ENTERPRISE_GRID_WPCOM,
 	PLAN_HOSTING_TRIAL_MONTHLY,
+	PlanSlug,
+	TERMS_LIST,
 } from '@automattic/calypso-products';
+import { AddOnMeta } from '@automattic/data-stores';
+import { UseCheckPlanAvailabilityForPurchase } from '@automattic/data-stores/src/plans';
 import { useMemo } from '@wordpress/element';
-import { GridPlan, PlansIntent } from '../../types';
+import { GridPlan, HiddenPlans, PlansIntent } from '../../types';
+import useGridPlans, { UseFreeTrialPlanSlugs } from './use-grid-plans';
 import useRestructuredPlanFeaturesForComparisonGrid from './use-restructured-plan-features-for-comparison-grid';
 
+const HIDDEN_PLANS = [ PLAN_HOSTING_TRIAL_MONTHLY, PLAN_ENTERPRISE_GRID_WPCOM ];
+
 interface Params {
 	allFeaturesList: FeatureList;
-	gridPlans: Omit< GridPlan, 'features' >[];
+	coupon?: string;
+	eligibleForFreeHostingTrial: boolean;
+	hiddenPlans: HiddenPlans;
 	intent?: PlansIntent;
+	isDisplayingPlansNeededForFeature: boolean;
+	isSubdomainNotGenerated: boolean;
 	selectedFeature?: string | null;
+	selectedPlan?: PlanSlug;
 	showLegacyStorageFeature?: boolean;
+	siteId?: number | null;
```

Comment:
I'm not sure I agree 🤔 

I don't think the hook should know/care about the notion of a site being 'selected'. Perhaps I want to use this hook in a different context, where the prop is based on a site being 'filtered'. The hook doesn't care _how_ this value was obtained, only that it is a site ID.

This would also apply to some other props, such as `selectedFeature` and `selectedPlan` ie. remove the contextual knowledge and just use `feature` and `plan`.

Wdyt?

Reply:
> Wdyt?

So I partly agree and disagree on your comment. 

I certainly do not agree that `selectedFeature` should be renamed to `feature`. `features` is a concept used throughout the grid, and the context to identify a given feature as "selected" is important. This: `<FeaturesGrid ... feature={ someSelectedFeature } />` is ambiguous to me. Unless `FeaturesGrid` is a super simple component, which is not, then I would not be happy with having to reason with `feature` being passed around into components that way and referring to something that's "highlighted" or "selected" or "hidden" (in the same way that I'd rather not have "plan" passed into the grid and be referring to "hiddenPlan").

For `siteId`, I would agree and disagree. I've actually been replacing a couple of cases in recent refactor from `siteId` to `selectedSiteId`. I had a couple of motivations behind it. 

First, these grids are part of WPCOM (they are not meant to be used outside), and the concept of a "selected site id" is a unique one - meaning, siteId is more general i.e. a user can have multiple sites, but only one selected at a time. 

Second, consistency with the underlying state/selectors. If something comes from `getSelectedSiteId` I'd rather name it `selectedSiteId`, in the same way that I'd prefer any `foo` to be coming from a `getFoo` selector. So unless we can rename the selector to `getSiteId`, then I'd rather usages to continue reflecting the name. I also think that big/broad components like the grids having knowledge of the "multisite" nature of the environment (that many sites exist for the user, with one being "selected"), then it's probably for the better.

Reply:
Yes I think you're right, `selectedFeature` and `selectedPlan` make sense. 

> First, these grids are part of WPCOM (they are not meant to be used outside), and the concept of a "selected site id" is a unique one - meaning, siteId is more general i.e. a user can have multiple sites, but only one selected at a time.

Only one site can be selected right now, however I can imagine a UI in WPCOM that allows users to configure multiple sites side-by-side. I understand this UI might not be practical for various design/other reasons, but it does help to illustrate my thinking around where `selectedSiteId` makes sense vs where `siteId` makes sense:

```
Site = ({ siteId }) => { ... }

MultiSiteEditor = () => {
  selectedSiteId = getSelectedId()

  return (
    <>
      <Site siteId={selectedSiteId} />
      <Site siteId={selectedSiteId} />
      <Site siteId={selectedSiteId} />
      ...
```


> Second, consistency with the underlying state/selectors. If something comes from getSelectedSiteId I'd rather name it selectedSiteId, in the same way that I'd prefer any foo to be coming from a getFoo selector. 

I agree that `selectedSiteId = getSelectedSiteId()` makes sense at the hook/selector call site, however not necessarily when that value is passed into other hooks or components (as per the eg I above)./

Wdyt? If you're not convinced, happy to adopt the strategy of 'selectedXx' everywhere. There's value in not having to think about it too much :)

Reply:
@oswian As you like. `siteId` sounds good. 

I'll find a way and re-refactor the other ones down the line too.

Reply:
> I'll find a way and re-refactor the other ones down the line too.

I've renamed other instances in `plans-grid-next` and `.../my-sites/plans-features-main/index.tsx`: https://github.com/Automattic/wp-calypso/pull/87478/commits/9b7df3e3fad7dad9a851dcf27673bf14d64042fe